src/data/set/basic.lean

```diff
@@ -892,6 +892,11 @@ set.ext $ λ _, or_and_distrib_left
 theorem union_inter_distrib_right {s t u : set α} : (s ∪ t) ∩ u = (s ∩ u) ∪ (t ∩ u) :=
 set.ext $ λ _, or_and_distrib_right
 
+theorem inter_union_compl (α:Type*) (A B:set α):(A∩ B)∪ (A∩ Bᶜ)=A :=
+begin
+  rw ← inter_union_distrib_left, rw union_compl_self, simp,
+end
```

Comment:
I believe this lemma is the same as `inter_union_diff`, however I think this one is still nice to have since `rw` can't see through defeq. If you move this one below `inter_union_diff`, then you can use

```lean
theorem inter_union_compl (s t : set α) : (s ∩ t) ∪ (s ∩ tᶜ) = s := inter_union_diff _ _
```

Reply:
Also I used `s` and `t` as the name for the sets since that seems to be the convention used in this file.

Reply:
Refactored as suggested.